src/honey_badger/honey_badger.rs

```diff
@@ -24,14 +25,92 @@ pub struct HoneyBadger<C, N: Rand> {
     pub(super) epochs: BTreeMap<u64, EpochState<C, N>>,
     /// The maximum number of `Subset` instances that we run simultaneously.
     pub(super) max_future_epochs: u64,
-    /// Messages for future epochs that couldn't be handled yet.
-    pub(super) incoming_queue: BTreeMap<u64, Vec<(N, MessageContent<N>)>>,
+    /// Messages that couldn't be handled yet by remote nodes.
+    pub(super) outgoing_queue: BTreeMap<(N, u64), Vec<Message<N>>>,
     /// Known current epochs of remote nodes.
     pub(super) remote_epochs: BTreeMap<N, u64>,
 }
 
 pub type Step<C, N> = messaging::Step<HoneyBadger<C, N>>;
 
+impl<C, N> Step<C, N>
+where
+    C: Contribution + Serialize + for<'r> Deserialize<'r>,
+    N: NodeIdT + Rand,
+{
+    /// Removes and returns any messages that are not accepted by remote nodes according to the
+    /// mapping `remote_epochs`. This way the returned messages are postponed until later, and the
+    /// remaining messages can be sent to remote nodes without delay.
+    fn defer_messages<'i, I>(
+        &mut self,
+        remote_epochs: &'i BTreeMap<N, u64>,
+        max_future_epochs: u64,
+        remote_ids: I,
+    ) -> impl Iterator<Item = (N, Message<N>)>
+    where
+        I: 'i + Iterator<Item = &'i N>,
+        N: 'i,
+    {
+        let accepts = |us: u64, them: u64| them <= us && us <= them + max_future_epochs;
+        let is_early = |us: u64, them: u64| us < them;
```

Comment:
Should this be called `is_late` or `is_outdated`? Those messages can probably be discarded right away?